ios/RNSScreen.m

```diff
@@ -142,6 +148,54 @@ - (void)setStackAnimation:(RNSScreenStackAnimation)stackAnimation
   }
 }
 
+- (void)setStatusBarStyle:(RNSStatusBarStyle)statusBarStyle
+{
+  if (statusBarStyle != _statusBarStyle) {
+    if (![[[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"] boolValue]) {
+      RCTLogError(@"If you want to change the style of status bar, you have to change \
+      UIViewControllerBasedStatusBarAppearance key in the Info.plist to YES");
+    } else {
+      _statusBarStyle = statusBarStyle;
+      _controller.modalPresentationCapturesStatusBarAppearance = YES;
+      [UIView animateWithDuration:0.5 animations:^{
+        [self->_controller setNeedsStatusBarAppearanceUpdate];
+      }];
+    }
+  }
+}
+
+- (void)setStatusBarAnimation:(UIStatusBarAnimation)statusBarAnimation
+{
+  if (statusBarAnimation != _statusBarAnimation) {
+    if (![[[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"] boolValue]) {
+      RCTLogError(@"If you want to change the animation of status bar, you have to change \
+      UIViewControllerBasedStatusBarAppearance key in the Info.plist to YES");
+    } else {
+      _statusBarAnimation = statusBarAnimation;
+      _controller.modalPresentationCapturesStatusBarAppearance = YES;
+      [UIView animateWithDuration:0.5 animations:^{
+        [self->_controller setNeedsStatusBarAppearanceUpdate];
+      }];
+    }
+  }
+}
+
+- (void)setStatusBarHidden:(BOOL)statusBarHidden
+{
+  if (statusBarHidden != _statusBarHidden) {
+    if (![[[NSBundle mainBundle] objectForInfoDictionaryKey:@"UIViewControllerBasedStatusBarAppearance"] boolValue]) {
```

Comment:
Maybe extract this to a separate method? We don't need special warning per parameter. Alternatively it'd be the best to evaluate objectForInfoDictionaryKey call in global scope and store it in a const, mainBundle cannot be changed in runtime so it should be safe to only evaluate it this way